stablehlo/dialect/StablehloOps.cpp

```diff
@@ -3363,15 +3363,15 @@ LogicalResult SetDimensionSizeOp::inferReturnTypes(
 // PadOp
 //===----------------------------------------------------------------------===//
 
-LogicalResult PadOp::inferReturnTypeComponents(
-    MLIRContext*, Optional<Location> location, ValueShapeRange operands,
+LogicalResult PadOp::inferReturnTypes(
```

Comment:
If the assert is the only issue, the correct way would be to remove that upstream and just use the attribute in building RankedTensorType. It wasn't used so far and therefore the assert was added. 

I am okay to make progress in parallel or simply wait on that to be fixed. Not sure how long it takes for StableHLO LLVM upgrade.

Reply:
Addressing this upstream is a good thing to do, but to me it looks like a lower priority than making progress with shape functions, given that they are needed both for the TF/XLA bridge and JAX, whereas inferReturnTypes vs inferReturnTypeComponents doesn't seem to make much difference at this point.

Reply:
Like Eugene states, we do not have strong preference to use which interface and there's no reason to be blocked by that in any way. So let's move on with this. And only consider fix the upstream when we have a stronger or higher priority motivation.